### PR TITLE
customize port/rootURL in http mode

### PR DESCRIPTION
**Pull request by levb**

### CODE REVIEW

File: manifest.json [hunk 1]
```diff
@@ -25,9 +25,6 @@
             }
         ]
     },
-    "http": {
-        "root_url": "http://localhost:4445"
-    },
```
Comment (mickmister) [inline manifest.json:30]: I feel that removing this from the manifest may signal to a developer that the app does not support running in http mode. I think inspecting an app's manifest is a good way to determine the deployment methods it supports.

**Review by mickmister — APPROVED** [review]
LGTM :+1: Just one suggestion on keeping the http mode info in the manifest, even though it is overwritten at runtime.

File: http-server/main.go [hunk 2]
```diff
@@ -2,22 +2,38 @@ package main
...
 
+	"github.com/mattermost/mattermost-plugin-apps/apps"
 	"github.com/mattermost/mattermost-plugin-apps/utils"
 
+	root "github.com/mattermost/mattermost-app-servicenow"
 	"github.com/mattermost/mattermost-app-servicenow/function"
 )
 
 func main() {
+	portStr := os.Getenv("PORT")
```
Comment (hanzei) [inline http-server/main.go:18]: Should these env variables be documented in the readme?

File: manifest.json (hunk 1 above)
Comment (levb) [reply manifest.json:30]: done

File: http-server/main.go (hunk 2 above)
Comment (levb) [reply http-server/main.go]: done. also updated the readme and docs in general

File: http-server/main.go [hunk 3]
```diff
@@ -2,22 +2,42 @@ package main
...
+
 	r := mux.NewRouter()
 	log := utils.MustMakeCommandLogger(zapcore.DebugLevel)
 	function.Init("http", r, log)
 	http.Handle("/", r)
 
-	listen := ":4445"
-	log.Infof("servicenow app started, manifest at http://localhost%s/manifest.json", listen)
+	listen := ":" + portStr
+	log.Infof("servicenow app started, listening on port %s, manifest at %s/manifest.json", portStr, rootURL)
```
Comment (hanzei) [inline http-server/main.go:41]: Does this correctly show if rootURL is set via an env variable?

File: http-server/main.go [hunk 4]
```diff
@@ -2,22 +2,42 @@ package main
...
+
+	portStr := os.Getenv("PORT")
+	if portStr == "" {
+		u, err := url.Parse(root.AppManifest.Deploy.HTTP.RootURL)
+		if err != nil {
+			panic(err)
+		}
+		portStr = u.Port()
+		if portStr == "" {
+			portStr = "8080"
```
Comment (mickmister) [inline http-server/main.go:31]: This port should match the port in `manifest.json`, or vise versa

Comment (levb) [reply http-server/main.go:31]: It does. 8080 is the default if manifest.json doesn't contain a port (and there's no PORT env)

File: http-server/main.go (hunk 3 above)
Comment (levb) [reply http-server/main.go:41]: image
and
image

Comment (levb) [reply http-server/main.go:41]: image

image

**Comment by levb** [comment]
This does not affect the production deployment, all changes are for HTTP dev environments

**Review by hanzei — APPROVED** [review]
LGTM

**Pull request merged**